### PR TITLE
[Merged by Bors] - feat: introduce fluvio hub util protocol

### PR DESCRIPTION
**Pull request by EstebanBorai**
Splits logic from taxonomy for the Fluvio Hub Util crate by disengaging
type definitions from logic and providing a `Ext`-ish trait to extend
`PackageMeta` and other types behavior.

### CODE REVIEW

**Review by sehz — APPROVED** [review]
Assumes that this is a just separation

File: crates/fluvio-hub-util/src/hubaccess.rs [hunk 1]
```diff
@@ -6,12 +6,12 @@ use surf::http::mime;
 use surf::StatusCode;
 use tracing::{debug, info};
 
+use fluvio_hub_protocol::{Result, HubUtilError};
```
Comment (digikata) [inline crates/fluvio-hub-util/src/hubaccess.rs:9]: The HubUtilError naming for the error types seems like it should maybe be updated to something else. Maybe just HubError?

Comment (EstebanBorai) [reply crates/fluvio-hub-util/src/hubaccess.rs]: Addressed on: [2e6191d](https://github.com/infinyon/fluvio/pull/2841/commits/2e6191d8a46f00d69256f3bd9a8ba91869e49c4c)

**Review by digikata — APPROVED** [review]
LGTM

**Pull request closed without merging**